contractsV5/KyberNetwork.sol

```diff
@@ -508,7 +508,8 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
         uint totalFeeWei;
         uint tradeWeiAmount;
         uint destAmount;
-        uint rate; //the accumulated rate for full trade
+        uint rateWithFee; // rate for full trade, after accounting for fees
```

Comment:
Seems to me we can have:
destAmountWithFee
destAmountNoFee

or 
like here.
no need for both.

In general the code now works with amounts.
if we move to rates. need to check it works nice in other parts of the code